src/vos/vos_iterator.c

```diff
@@ -683,11 +681,18 @@ vos_iterate_internal(vos_iter_param_t *param, vos_iter_type_t type,
 			if (acts & VOS_ITER_CB_ABORT)
 				break;
 
+			if (acts & VOS_ITER_CB_RESTART) {
+				daos_anchor_set_zero(anchor);
+				probe_anchor = NULL;
+				goto probe;
```

Comment:
Simply re-probe won't work for sorted evtree iteration, you need to re-prepare the sorted entries on restart, since the deletion before restart have invalidated the visibility of evtree entries.

I think a simpler way is to rely on next round of aggregation to aggregate the same epoch range again. (we just need to skip updating HAE when aggregation aborted on hitting aborted or uncommitted entries.)

Reply:
You don't need to re-prepare.   For sorted iterator, probe is what re-reads all the entries and sorts them so it works to simply re-probe in this case.   See evt_iter_probe_sorted.  I suppose it may be a bit weird for upper levels but it's only really used by evtree so maybe it's fine.  I originally did re-prepare but this actually requires a reprobe of the parent iterator and is more complicated.

We could do as you say and just abort but what is the advantage of doing that?

Reply:
Ok, I was thinking reprobe won't re-read all entries.